tests/test_efs/test_file_system.py

```diff
@@ -0,0 +1,384 @@
+    desc_resp = efs.describe_file_systems()
+    desc_fs_resp_metadata = desc_resp.pop("ResponseMetadata")
+    assert desc_fs_resp_metadata["HTTPStatusCode"] == 200
+
+    # Check the list results.
+    fs_list = desc_resp["FileSystems"]
+    assert len(fs_list) == 1
+    file_system = fs_list[0]
+
+    assert set(file_system.keys()) == {
+        "AvailabilityZoneId",
+        "AvailabilityZoneName",
+        "CreationTime",
+        "CreationToken",
+        "Encrypted",
+        "LifeCycleState",
+        "PerformanceMode",
+        "ProvisionedThroughputInMibps",
+        "SizeInBytes",
+        "Tags",
+        "ThroughputMode",
+        "FileSystemId",
+        "FileSystemArn",
+        "NumberOfMountTargets",
+        "OwnerId",
+    }
+    assert file_system["ProvisionedThroughputInMibps"] == 60
+    assert file_system["AvailabilityZoneId"] == "usw2-az1"
+    assert file_system["AvailabilityZoneName"] == "us-west-2b"
+    assert file_system["ThroughputMode"] == "provisioned"
+
+
+def test_describe_file_systems_paging(efs):
+    # Create several file systems.
+    for i in range(10):
+        efs.create_file_system(CreationToken="foobar_{}".format(i))
+
+    # First call (Start)
+    # ------------------
+
+    # Call the tested function
+    resp1 = efs.describe_file_systems(MaxItems=4)
+
+    # Check the response status
+    assert has_status_code(resp1, 200)
+
+    # Check content of the result.
+    resp1.pop("ResponseMetadata")
+    assert set(resp1.keys()) == {"NextMarker", "FileSystems"}
+    assert len(resp1["FileSystems"]) == 4
+    fs_id_set_1 = {fs["FileSystemId"] for fs in resp1["FileSystems"]}
+
+    # Second call (Middle)
+    # --------------------
+
+    # Get the next marker.
+    resp2 = efs.describe_file_systems(MaxItems=4, Marker=resp1["NextMarker"])
+
+    # Check the response status
+    resp2_metadata = resp2.pop("ResponseMetadata")
+    assert resp2_metadata["HTTPStatusCode"] == 200
+
+    # Check the response contents.
+    assert set(resp2.keys()) == {"NextMarker", "FileSystems", "Marker"}
+    assert len(resp2["FileSystems"]) == 4
+    assert resp2["Marker"] == resp1["NextMarker"]
+    fs_id_set_2 = {fs["FileSystemId"] for fs in resp2["FileSystems"]}
+    assert fs_id_set_1 & fs_id_set_2 == set()
+
+    # Third call (End)
+    # ----------------
+
+    # Get the last marker results
+    resp3 = efs.describe_file_systems(MaxItems=4, Marker=resp2["NextMarker"])
+
+    # Check the response status
+    resp3_metadata = resp3.pop("ResponseMetadata")
+    assert resp3_metadata["HTTPStatusCode"] == 200
+
+    # Check the response contents.
+    assert set(resp3.keys()) == {"FileSystems", "Marker"}
+    assert len(resp3["FileSystems"]) == 2
+    assert resp3["Marker"] == resp2["NextMarker"]
+    fs_id_set_3 = {fs["FileSystemId"] for fs in resp3["FileSystems"]}
+    assert fs_id_set_3 & (fs_id_set_1 | fs_id_set_2) == set()
+
+
+def test_describe_file_systems_invalid_marker(efs):
+    try:
```

Comment:
Moto tends to use pytest.raises to verify error scenarios, which is a bit cleaner IMO:
```
with pytest.raises(ClientError) as ex:
    ..
err = ex.value.response["Error"]
err["Message"].should.equal(..)
err["Code"].should.equal(..)
```

This should be applied for all every negative test.

Reply:
> Moto tends to use pytest.raises to verify error scenarios, which is a bit cleaner IMO:
> 
> ```
> with pytest.raises(ClientError) as ex:
>     ..
> err = ex.value.response["Error"]
> err["Message"].should.equal(..)
> err["Code"].should.equal(..)
> ```
> 
> This should be applied for all every negative test.

Alright, this should be all set now. Thanks for the pointer!